nixos/tests/systemd-timesyncd-nscd-dnssec.nix

```diff
@@ -0,0 +1,61 @@
+# This test verifies that systemd-timesyncd can resolve the NTP server hostname when DNSSEC validation
+# fails even though it is enforced in the systemd-resolved settings. It is required in order to solve
+# the chicken-and-egg problem when DNSSEC validation needs the correct time to work, but to set the
+# correct time, we need to connect to an NTP server, which usually requires resolving its hostname.
+#
+# This test does the following:
+# - Sets up a DNS server (tinydns) listening on the eth1 ip addess, serving .ntp and fake.ntp records.
+# - Configures that DNS server as a resolver and enables DNSSEC in systemd-resolved settings.
+# - Configures systemd-timesyncd to use fake.ntp hostname as an NTP server.
+# - Executes getent command to make sure DNS lookup fails due to broken DNSSEC.
```

Comment:
This still mentions getent. Maybe just "performs a regular DNS lookup, to ensure it fails due to broken DNSSEC"?

Reply:
Good catch. Fixed.